### PR TITLE
Logging tests

### PR DESCRIPTION
**Pull request by achaikou**

### CODE REVIEW

File: cmd/query/utils_main_test.go [hunk 1]
```diff
@@ -182,35 +191,101 @@ func prepareRequest(ctx *gin.Context, t *testing.T, endpoint string,
...
-func prepareMetadataRequest(ctx *gin.Context, t *testing.T, testcase metadataTest) {
-	prepareRequest(
-		ctx,
-		t,
-		"/metadata",
-		testcase.method,
-		testcase.metadata,
-		testcase.metadataJSONRequest,
-	)
+//---------- SliceTest----------------
```
[the remark this answers is not in the document]
Comment (achaikou) [reply cmd/query/utils_main_test.go]: Deleted.

File: cmd/query/utils_main_test.go [hunk 2]
```diff
@@ -182,35 +191,101 @@ func prepareRequest(ctx *gin.Context, t *testing.T, endpoint string,
...
+func (s sliceTest) getName() string {
```
[the remark this answers is not in the document]
Comment (achaikou) [reply cmd/query/utils_main_test.go]: Changed to something similar to what we discussed, but not sure if you'd deem it good enough.
But now it should look more go-ish than java-ish, so I like it more.

File: cmd/query/main.go [hunk 3]
```diff
@@ -126,25 +126,7 @@ func parseopts() opts {
...
-
-	storageAccounts := strings.Split(opts.storageAccounts, ",")
-	
-	endpoint := api.Endpoint{
-		MakeVdsConnection: vds.MakeAzureConnection(storageAccounts),
-		Cache:             cache.NewCache(opts.cacheSize),
-	}
-
-	app := gin.New()
+func setupRouting(app *gin.Engine, endpoint *api.Endpoint, opts *opts) {
```
Comment (ErlendHaa) [inline cmd/query/main.go:129]: This comment is a bit misplaced due to GitHub.

I don't think setting up  and running the metrics app fits in this function. This function, as the name suggests, should only be responsible for adding the right routes and middleware to the main app. 

You could make another function for the metrics app, (Essentially everything in the `if opts.metrics` expect the 2 first lines), but I would just inline that in main unless we need the same setup for testing. 

Also I would prefer that we pass in the variables from opts that we need, not the entire opts struct

Comment (achaikou) [reply cmd/query/main.go]: Mostly agree, but ended up with a bit different version. I don't exactly like it, but I can't pinpoint what I dislike and how to fix it, so this seems to me the best option.

**Review by ErlendHaa — APPROVED** [review]
LGTM

**Pull request merged**